internal/internal_task_handlers.go

```diff
@@ -917,12 +924,19 @@ ProcessEvents:
 		}
 	}
 	if nonDeterministicErr == nil && w.err != nil {
-		if isReplayTest {
-			// NOTE: we should check the following error regardless if it's in replay test or not
-			// but since we are not checking it previously, it may break existing customers workflow
-			if panicErr, ok := w.err.(*workflowPanicError); ok && panicErr.value != nil {
-				if _, isStateMachinePanic := panicErr.value.(stateMachineIllegalStatePanic); isStateMachinePanic {
+		if panicErr, ok := w.err.(*workflowPanicError); ok && panicErr.value != nil {
+			if _, isStateMachinePanic := panicErr.value.(stateMachineIllegalStatePanic); isStateMachinePanic {
```

Comment:
What are the other possible cases? Why not log for all?

Reply:
Only `stateMachineIllegalStatePanic` error is counted as non-deterministic error. 

Other cases may be panic in the workflow code that is caught by the library. For those cases, there's another check (with logging) when completing decision task and responding decision task failed.